pkg/controller/common/util.go

```diff
@@ -602,15 +631,38 @@ func ValidateCloneTokenDV(validator token.Validator, dv *cdiv1.DataVolume) error
 		return errors.Wrap(err, "error verifying token")
 	}
 
-	return validateTokenData(tokenData, dv.Spec.Source.PVC.Namespace, dv.Spec.Source.PVC.Name, dv.Namespace, dv.Name, "")
+	tokenResourceName := getTokenResourceNameDataVolume(dv.Spec.Source)
+	if tokenResourceName == "" {
+		return errors.New("token resource name empty, can't verify properly")
+	}
+
+	return validateTokenData(tokenData, sourceNamespace, sourceName, dv.Namespace, dv.Name, "", tokenResourceName)
+}
+
+func getTokenResourceNameDataVolume(source *cdiv1.DataVolumeSource) string {
+	if source.PVC != nil {
+		return "persistentvolumeclaims"
+	} else if source.Snapshot != nil {
+		return "volumesnapshots"
+	}
+
+	return ""
+}
+
+func getTokenResourceNamePvc(sourcePvc *corev1.PersistentVolumeClaim) string {
+	if v, ok := sourcePvc.Labels[common.CDIComponentLabel]; ok && v == common.CloneFromSnapshotFallbackPVCCDILabel {
+		return "volumesnapshots"
+	}
+
+	return "persistentvolumeclaims"
 }
```

Comment:
using (typed?) consts might be better here for "persistentvolumeclaims" and "volumesnapshots"

Reply:
so I am struggling to find these plural names in API libs and feels a bit wrong to define them ourselves,
any idea what I am missing?
We use `"persistentvolumeclaims"` quite a bit outside this file as well, so I am assuming someone ran into this before me